diff --git a/src/Plugin/better_exposed_filters/filter/SelectiveFilterBase.php b/src/Plugin/better_exposed_filters/filter/SelectiveFilterBase.php
index 48792fa..3242793 100644
--- a/src/Plugin/better_exposed_filters/filter/SelectiveFilterBase.php
+++ b/src/Plugin/better_exposed_filters/filter/SelectiveFilterBase.php
@@ -185,11 +185,15 @@ abstract class SelectiveFilterBase {
               )
             ) {
               $element['#access'] = FALSE;
+              // Hide collapsible details element
+              unset($form[$identifier . '_collapsible']);
             }
           }
         }
         elseif (!empty($settings['options_hide_when_empty'])) {
           $element['#access'] = FALSE;
+          // Hide collapsible details element
+          unset($form[$identifier . '_collapsible']);
         }
       }
       else {
